comp/forwarder/defaultforwarder/go.mod

```
@@ -0,0 +1,152 @@
+module github.com/DataDog/datadog-agent/comp/forwarder/defaultforwarder
+
+go 1.21
```

I assume this is meant to be used with the Collector, until Go 1.21 is released this should be 1.20
```suggestion
go 1.20
```

---

For now, we are making modules using `go 1.21`. We are at least a couple of months away from getting to use our modules in the collector. The following steps would get us there: 

1. Add tests for `go 1.20`. 
2. Mark the appropriate modules in use by OTel in `tasks/modules.py` . 
3. Move  `pkg/trace/api/otlp.go` to opentelemetry-mapping-go .

---

Alright, no problem if it is intentional :)